src/clients/patreon/patreonAgent.ts

```diff
@@ -191,14 +186,14 @@ export class PatreonAgent {
         patreonRecordUpdateDetails,
       );
     } else if (hasPaid) {
+      // Create new patreonRecord only if hasPaid. Else do not create
       await patreonRecord.create(patreonRecordUpdateDetails);
     }
```

Comment:
Have a think about how we can best reorganise this. 

Maybe we should always store an update first, then selectively update the currentPledgeExpiryDate. 
Let's also not leave the else not there. In the worst case if we must, add an else and chuck the return in it.